server/src/main/java/org/elasticsearch/common/xcontent/ChunkedToXContentHelper.java

```diff
@@ -74,6 +74,20 @@ public static Iterator<ToXContent> field(String name, String value) {
         return Iterators.single(((builder, params) -> builder.field(name, value)));
     }
 
+    /**
+     * Creates an Iterator to serialize a named field where the value is represented by a chunked ToXContext.
+     * Chunked equivalent for {@code XContentBuilder field(String name, ToXContent value)}
+     * @param name name of the field
+     * @param value ChunkedToXContent value for this field (single value, object or array)
+     * @return Iterator composing field name and value serialization
+     */
+    public static Iterator<ToXContent> field(String name, ChunkedToXContent value) {
+        return Iterators.concat(
+            Iterators.single((builder, params) -> builder.field(name)),
+            value.toXContentChunked(ToXContent.EMPTY_PARAMS)
```

Comment:
I think we need to propagate the params from the caller rather than using `EMPTY_PARAMS` here. It might well not matter right now but it will catch us out in future.

Reply:
If you think it's safer, I will change it. 
The reason why I used `EMPTY_PARAMS` is that the original code calls into this `XContentBuilder` [function](https://github.com/elastic/elasticsearch/blob/2df6def54fbff76a1d0108cbef7d9dca46757741/libs/x-content/src/main/java/org/elasticsearch/xcontent/XContentBuilder.java#L963), which calls [this function](https://github.com/elastic/elasticsearch/blob/2df6def54fbff76a1d0108cbef7d9dca46757741/libs/x-content/src/main/java/org/elasticsearch/xcontent/XContentBuilder.java#L971) with `EMPTY_PARAMS`. 
Should I provide both overloads, or maybe just the one with `params` to encourage people to propagate them?

Reply:
I'd prefer just the one with params so that people have to propagate. The one without params in `XContentBuilder` is equally trappy IMO.

Reply:
Makes sense, done!